src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/ApprovedPremisesRoomsSeedFromXLSXJob.kt

```diff
@@ -0,0 +1,166 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.cas1
+
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.api.getColumn
+import org.jetbrains.kotlinx.dataframe.name
+import org.slf4j.LoggerFactory
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ServiceName
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BedEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BedRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PremisesEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PremisesRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.RoomEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.RoomRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.ExcelSeedJob
+import java.util.UUID
+
+class ApprovedPremisesRoomsSeedFromXLSXJob(
+  fileName: String,
+  premisesId: UUID,
+  sheetName: String,
+  private val premisesRepository: PremisesRepository,
+  private val roomRepository: RoomRepository,
+  private val bedRepository: BedRepository,
+  private val characteristicRepository: CharacteristicRepository,
+  private val siteSurvey: SiteSurvey,
+) : ExcelSeedJob(
+  fileName = fileName,
+  premisesId = premisesId,
+  sheetName = sheetName,
+) {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  override fun processDataFrame(dataFrame: DataFrame<*>) {
+    findExistingPremisesOrThrow(premisesId)
+
+    val questions = dataFrame.getColumn(0).values().toList()
+    if (questions.drop(2) != siteSurvey.questionToCharacterEntityMapping.keys.toList()) throw RuntimeException("Site survey file $fileName contains wrong questions on sheet $sheetName")
+
+    for (i in 1..<dataFrame.columnsCount()) {
+      val roomCharacteristics = dataFrame.getColumn(i)
+      val roomCode = roomCharacteristics.name
+      val roomName = roomCharacteristics[0].toString()
+
+      val room = roomRepository.findByCode(roomCode) ?: createRoom(
+        roomCode,
+        roomName,
+      )
+
+      val bedCode = roomCharacteristics[1].toString()
+      val bedName = "$roomName - $bedCode"
```

Comment:
I think we should be using the value of the 'Room Number / Name' row for this. I _think_ this is the first time premises have been able to define their own names